test/e2e/nodeadm.go

```diff
@@ -15,10 +15,17 @@ import (
 type NodeadmOS interface {
 	Name() string
 	AMIName(ctx context.Context, awsConfig aws.Config) (string, error)
-	BuildUserData(UserDataInput UserDataInput) ([]byte, error)
-	InstanceType(region string) string
+	BuildUserData(userDataInput UserDataInput) ([]byte, error)
+	InstanceType(region string, instanceSize InstanceSize) string
 }
 
+type InstanceSize string
```

Comment:
nit if we are only using for comparison we can use an int with `iota`, slightly more memory efficient